src/main.rs

```diff
@@ -686,6 +688,27 @@ impl TokenSink for TokenPrinter {
                         "tr" => {
                             self.is_table_row = Some(Vec::new());
                         }
+                        // HACK: spans are only supported enough to get syntax highlighting in code
+                        // blocks working
+                        "span" => {
+                            if let Some(Attribute { name, value }) = tag.attrs.get(0) {
+                                if &name.local == "style" {
+                                    let styles = value.to_string();
+                                    if let Some(hex_str) = styles
+                                        .split(';')
+                                        .find_map(|style| style.strip_prefix("color:#"))
+                                    {
+                                        if let Ok(hex) = u32::from_str_radix(hex_str, 16) {
+                                            let [_, r, g, b] = hex.to_be_bytes();
+                                            let r = r as f32 / 255.0;
+                                            let g = g as f32 / 255.0;
+                                            let b = b as f32 / 255.0;
+                                            self.span_color = [r, g, b, 1.0];
```

Comment:
Using the function `hex_to_linear_rgba` from colour.rs like `self.state.span_color = hex_to_linear_rgba(hex);` helps to convert the srgb web colour into rgba linear which wgpu uses. We need to do this conversion to stop colors washing out and appearing muted.